### PR TITLE
GEODE-2437 Add travis-ci file

### PR DESCRIPTION
**Pull request by metatype**
Configure travis-ci to build cppcache on linux and run unit
tests.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  - sudo apt-get -qq update
+  - sudo apt-get -y build-essential
+  - sudo apt-get install -y cmake
+  - sudo apt-get install -y doxygen
+  - sudo apt-get install -y git
+  - sudo apt-get install -y openjdk-8-jdk
+  - sudo apt-get install -y wget
+  - sudo apt-get install -y zlib1g-dev
+
+install: mkdir build && cd build && cmake ../src && cmake --build . -- -j 8
```
Comment (echobravopapa) [inline .travis.yml]: The configuration step will need to be like so: 
`make ../src -DGEODE_ROOT=<pathtogeode> `
optionally one could add:
`-DCMAKE_BUILD_TYPE=Debug` Debug is default unless Release is specified
`-DPRODUCT_VERSION=maj.min.patch-build.N`

Comment (metatype) [inline .travis.yml]: Do the unit tests require a GEODE install?

Comment (PivotalSarge) [inline .travis.yml]: I don't believe the unit tests require GEODE to run. They certainly shouldn't.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+language: cpp
+sudo: required
+
+before_install:
+  - sudo apt-get -qq update
+  - sudo apt-get -y build-essential
```
Comment (dgkimura) [inline .travis.yml]: We need GCC 4.8+ for full C++11 support.  Build #1992 looks like it has GCC 4.6.  I think the easiest way to fix is to add `dist: trusty`.

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  - sudo apt-get install -y cmake
```
Comment (dgkimura) [inline .travis.yml]: What CMake version does this install?  We need version 3.4 or higher.  If that's not the case, then we may need to `wget` and install manually.

File: .travis.yml (hunk 1 above)
Comment (echobravopapa) [inline .travis.yml]: It is needed to build the javaobject(s)...

**Pull request closed without merging**